File: testbed_php/test/associations/ReflexiveOptionalOneTest.php
<?php

class ReflexiveOptionalOneTest extends UnitTestCase
{

  public function test_SetSuperMentor()
  {
    $m = new MentorH("m1");
    $m2 = new MentorH("m2");
    
    $m->setSuperMentor($m2);
    $this->assertEqual($m, $m2->getSuperMentor());
    $this->assertEqual($m2, $m->getSuperMentor());
  }
  
}

File: testbed_python/test/associations/reflexive_optional_one_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["MentorH"], ["associations"])
from ImportModules import *


class ReflexiveOptionalOneTest(unittest.TestCase):
    def test_SetSuperMentor(self):
        m = MentorH.MentorH("m1")
        m2 = MentorH.MentorH("m2")

        m.setSuperMentor(m2)
        self.assertEqual(m, m2.getSuperMentor())
        self.assertEqual(m2, m.getSuperMentor())
